int16 mode
uint16 msecs_to
---
int8 confirm